### PR TITLE
refactor: Apply Model-View-Presenter pattern to about dialog

### PR DESCRIPTION
**Pull request by MattHag**
Related #2395

### CODE REVIEW

**Comment by pfps** [comment]
I really don't see any rationale for replacing about 80 lines of code with more than twice that including lots of ... bits and methods that are never going to be needed.

**Comment by MattHag** [comment]
Yeah, I also thought that at first and knew it's not the prime example to start with. However, it's a nice little showcase of how the MVP pattern works and the remaining UI code should look like. The functions can be merged together, I don't care too much about changes here. But the restructuring achieves the gain here. The code is no longer tightly coupled to Gtk.

There are several important benefits, that outweigh the shorter code by far

- [ ] Unit tests are independent from Gtk. Gtk is a dependency for the view module, but not anywhere else.
- [ ] Drastically improves testability with unit tests. The model is easy, sure. But even the whole module can be tested with unit tests by replacing the UI module alone.
- [ ] Removes need of globals, which is a pain point for any unit test and should be replaced wherever possible
- [ ] Common pattern, used for all GUIs in the future

**Comment by pfps** [comment]
There is no update in the about dialog.  It simply presents information.  So I see no reason whatsoever to have there be anything but the construction of the window and its display.

One thing that could be done is to use a GUI builder, which would cut down on the required code.

**Comment by pfps** [comment]
The places that might benefit from a redesign are in the main window.   But the biggest parts there are heavily dependent on data structures.  One of these - the devices pane - is constructed on the fly.  The other - underlying the config panel - is the settings data for devices.

**Comment by MattHag** [comment]
The rule window is as important. Doesn't matter how it's built. If there's too much magic involved, then it just needs a refactoring. 
With all code smells like local imports and tight coupling still existing, we wouldn't even have a single test case running now. It all needs work, until I understand all the pieces and it's easy to maintain.

**Comment by pfps** [comment]
Agreed.  The rule window is also something that should be looked into.  

It again is a reflection of a data structure - here the Solaar rules.

File: lib/solaar/ui/about/model.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+##
+## You should have received a copy of the GNU General Public License along
+## with this program; if not, write to the Free Software Foundation, Inc.,
+## 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from solaar.i18n import _
+
+
+class AboutModel:
+    def __init__(self) -> None:
```
Comment (MattHag) [inline lib/solaar/ui/about/model.py:21]: Remove

**Comment by pfps** [comment]
I don't see the need for this change.  The previous code was simple.

**Comment by MattHag** [comment]
... and uses unnecessary globals, is not unit testable and has no clear structure, which makes it "simple". Understandable, testable and maintainable code will not be spaghetti code most of the time.

Only this so allows completely independent development of the UI, completely independent from any other part of Soolar :)

**Comment by pfps** [comment]
I don't see any problem with about.py and no need to change it.

**Pull request closed without merging**